zerver/actions/user_settings.py

```diff
@@ -488,19 +530,7 @@ def do_change_user_setting(
 
     # Updates to the time zone display setting are sent to all users
     if setting_name == "timezone":
-        payload = dict(
-            email=user_profile.email,
-            user_id=user_profile.id,
-            timezone=canonicalize_timezone(user_profile.timezone),
-        )
-        timezone_event = dict(type="realm_user", op="update", person=payload)
-        transaction.on_commit(
-            lambda: send_event(
-                user_profile.realm,
-                timezone_event,
-                active_user_ids(user_profile.realm_id),
-            )
-        )
+        send_timezone_update_event_to_all_users(user_profile)
```

Comment:
This leads to different events being sent to different emails based on email access. `email` field is already marked as `Depreceated` in the docs, so we can probably try to avoid this change. One workaround can be we do not update the user's `email` field in `apply_event` code using the email value passed in this event, since we already have different events to update email.